src/pkg/backup/details/details.go

```diff
@@ -141,13 +141,20 @@ func FolderEntriesForPath(parent, location *path.Builder) []folderEntry {
 	lfs := locationRefOf(location)
 
 	for len(parent.Elements()) > 0 {
-		nextParent := parent.Dir()
+		var (
+			nextParent = parent.Dir()
+			lr         string
+			dn         = parent.LastElem()
+		)
 
 		// TODO: We may have future cases where the storage hierarchy
 		// doesn't match the location hierarchy.
-		var lr string
 		if lfs != nil {
 			lr = lfs.String()
+
+			if len(lfs.Elements()) > 0 {
+				dn = lfs.LastElem()
+			}
```

Comment:
Not a blocker since we currently control inputs to this but... it seems like the behavior when we "bottom out" is that the stuff earlier in the path won't have display names correct? Since we're starting at the end (complete path) and removing elements as we go

Just, I think we need to make time to refactor this code in general 😅

Reply:
Agreed.  We def need a refactor, because we should be constructing folders by walking up the tree using only the location path, and not the full repo ref.  Ex: for every folder we're getting folder elems for the prefix (tenant/service/owner/type), which is wasteful, and not representative of anything outside of corso.  Also, it's supposed to be location by preference.

But before we do that, we need to ensure location is populated for all the data we produce.